### ISSUE TITLE
npm install error - installing web3@1.0.0-beta.26

### ISSUE THREAD

**Issue by LEscobar-Driver** [ethereum/web3.js#1561]
Ubuntu 16.04.4
npm -v : 5.6.0
node -v  9.10.1

cmd : npm install --save web3@1.0.0-beta.26

error message: 

`npm WARN deprecated fs-promise@2.0.3: Use mz or fs-extra^3.0 with Promise Support
npm WARN deprecated tar.gz@1.0.7: ⚠️  WARNING ⚠️ tar.gz module has been deprecated and your application is vulnerable. Please use tar module instead: https://npmjs.com/tar
scrypt@6.0.3 preinstall /home/ancientevil/CompleteDevelopersGuide/node_modules/scrypt
node node-scrypt-preinstall.js
scrypt@6.0.3 install /home/ancientevil/CompleteDevelopersGuide/node_modules/scrypt
node-gyp rebuild
Traceback (most recent call last):
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 13, in <module>
    import gyp
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 8, in <module>
    import gyp.input
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 13, in <module>
    import gyp.common
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/common.py", line 12, in <module>
    import tempfile
  File "/usr/lib/python2.7/tempfile.py", line 35, in <module>
    from random import Random as _Random
  File "/usr/lib/python2.7/random.py", line 49, in <module>
    import hashlib as _hashlib
  File "build/bdist.linux-x86_64/egg/hashlib.py", line 115, in <module>
    """
TypeError: 'frozenset' object is not callable
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at ChildProcess.emit (events.js:180:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:12)
gyp ERR! System Linux 4.13.0-38-generic
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/ancientevil/CompleteDevelopersGuide/node_modules/scrypt
gyp ERR! node -v v9.10.1
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm WARN webpack-cli@2.0.14 requires a peer of webpack@^4.0.0 but none is installed. You must install peer dependencies yourself.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! scrypt@6.0.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the scrypt@6.0.3 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.`

> End of the log file : 

6806 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules/fsevents):
6807 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
6808 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
6808 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
6808 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
6808 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: x64
6809 verbose stack Error: scrypt@6.0.3 install: `node-gyp rebuild`
6809 verbose stack Exit status 1
6809 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:283:16)
6809 verbose stack     at EventEmitter.emit (events.js:180:13)
6809 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
6809 verbose stack     at ChildProcess.emit (events.js:180:13)
6809 verbose stack     at maybeClose (internal/child_process.js:936:16)
6809 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:220:5)
6810 verbose pkgid scrypt@6.0.3
6811 verbose cwd /home/ancientevil/CompleteDevelopersGuide
6812 verbose Linux 4.4.0-119-generic
6813 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--save" "web3@1.0.0-beta.26"
6814 verbose node v9.8.0
6815 verbose npm  v5.8.0
6816 error code ELIFECYCLE
6817 error errno 1
6818 error scrypt@6.0.3 install: `node-gyp rebuild`
6818 error Exit status 1
6819 error Failed at the scrypt@6.0.3 install script.
6819 error This is probably not a problem with npm. There is likely additional logging output above.
6820 verbose exit [ 1, true ]

I have tried researching this but have found nothing that works. I have attempted to install this npm package on windows 7/10 & ubuntu to no avail. Extremely frustrating. I am unable to install web3 locally or globally.

**Comment by iurimatias** [comment]
@LEscobar-Driver did install the build-essential package in ubuntu ```sudo apt-get install build-essential``` first?

**Comment by KamuelaFranco** [comment]
Have you tried changing the Python version? I was able to install beta26 without an issue AFTER it correctly told me the build step only supported Python 2, and my anaconda install of python3 was blocking that.

I ran `npm config set python /usr/bin/python` to point it to my Mac's system version (2.7), and reinstalled without an issue.

I _believe_ this is the same issue, however the error message isn't as helpful when you're installing the latest beta.
